fragments/filter/README.md

```diff
@@ -0,0 +1,632 @@
+    - [Example 4](#example-4)
+      - [Example 4: AND cql-text (GET)](#example-4-and-cql-text-get)
+      - [Example 4: AND cql-json (POST)](#example-4-and-cql-json-post)
+    - [Example 5](#example-5)
+      - [Example 5: OR cql-text (GET)](#example-5-or-cql-text-get)
+      - [Example 5: OR cql-json (POST)](#example-5-or-cql-json-post)
+    - [Example 6: Temporal](#example-6-temporal)
+      - [Example 6: ANYINTERACTS cql-text (GET)](#example-6-anyinteracts-cql-text-get)
+      - [Example 6: ANYINTERACTS cql-json (POST)](#example-6-anyinteracts-cql-json-post)
+    - [Example 6: Spatial](#example-6-spatial)
+      - [Example 6: INTERSECTS cql-text (GET)](#example-6-intersects-cql-text-get)
+      - [Example 6: INTERSECTS cql-json (POST)](#example-6-intersects-cql-json-post)
+  - [Implementations](#implementations)
+
+## Overview
+
+The Filter extension provides an expressive mechanism for searching based on Item attributes.
+
+This extension uses several conformance classes defined in the 
+[OGC API - Features - Part 3: Filtering and the Common Query Language (CQL)](https://portal.ogc.org/files/96288)
+specification. As of May 2020, this specification is in draft status but, due to its long-standing use within 
+geospatial software (e.g., GeoServer), is expected to remain largely the same in final. 
+
+It should be noted that the "CQL" referred to here is OGC CQL. It is **not** referencing or related two other "CQL" languages, 
+the [SRU (Search/Retrieve via URL) Contextual Query Language](https://www.loc.gov/standards/sru/cql/index.html) (formerly 
+known as Common Query Language) or the [Cassandra Query Language](https://cassandra.apache.org/doc/latest/cql/) used by the Cassandra database.
+
+OGC CQL has been previously described 
+in [OGC Filter Encoding](https://www.ogc.org/standards/filter) and [OGC Catalogue Services 3.0 - General Model](http://docs.opengeospatial.org/is/12-168r6/12-168r6.html#62) (including a BNF grammar in Annex B). 
+OAFeat Part 3 CQL formally defines syntax for both a text format (cql-text) as an ABNF grammar (largely similar to the BNF grammar in the General Model) and a JSON format (cql-json) as an OpenAPI schema, and provides a precise natural language description of the declarative semantics.
+
+## Limitations of Item Search 
+
+OAFeat defines a limited set of filtering capabilities. Filtering can only be done over a single collection and 
+with only a single `bbox` (rectangular spatial filter) parameter and a single datetime (instant or interval) parameter. 
+
+The STAC Item Search specification extends the functionality of OAFeat in a few key ways:
+- It allows cross-collection filtering, whereas OAFeat only allows filtering within a single collection. (`collections` parameter, accepting 0 or more collections)
+- It allows filtering by Item ID (`ids` parameter)
+- It allows filtering based on a single GeoJSON Geometry, rather than only a bbox (`intersects` parameter)
```

Comment:
>  But the intersects parameter is definitely simpler, as you don't need to include the name of the geometry and figure out as much of how CQL works. 

Yes, this is where I stand. I think people should be about to implement intersects without even needing to touch queryables and CQL.

Reply:
Ok. I do think if that's our stance (which I'm cool with) we should at least try a bit to advocate to OGC to make it an extension for OGC API - Features. So instead of the 'intersects CQL class' we'd have an OGC parameter that is 'intersects'. For OGC API - Features they'd want that as an extension, but for STAC we'd make it a requirement.

Feel free to give that feedback, or I'm also happy to do so.

Reply:
I'll put that on my todo list.